source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1264,9 +1282,13 @@ implementation::
                 return
 
             if isStateChangeError(error):
-                # Mark the server Unknown
-                unknown = new ServerDescription(type=Unknown, error=error, topologyVersion=topologyVersion)
-                onServerDescriptionChanged(unknown, connection pool for server)
+                # Don't mark server unknown in load balanced mode.
+                if type != LoadBalanced
+                  # Mark the server Unknown
+                  unknown = new ServerDescription(type=Unknown, error=error, topologyVersion=topologyVersion)
+                  onServerDescriptionChanged(unknown, connection pool for server)
+                else
```

Comment:
This `else` statement is not correct - we only want to clear the pool for the specific `serviceId` on shutdown errors or pre-4.2 state change errors. That's already covered by the next `if` statement, but it needs to be amended to mention `serviceId`. I'd recommend removing this `else` and modifying the `if` to:

```
if isShutdown(code) or (error was from <4.2):
  if type == LoadBalanced:
    clear connection pool for serviceId
  else:
    clear connection pool for server
```

Reply:
This has now been updated.